src/main/antlr/org/hibernate/sqm/parser/hql/internal/antlr/HqlParser.g4

```diff
@@ -87,10 +88,26 @@ orderingSpecification
 	;
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// LIMIT/OFFSET clause
+
+limitClause
+	: LIMIT parameterOrNumberLiteral
+	;
+
+offsetClause
+	: OFFSET parameterOrNumberLiteral
+	;
+
+parameterOrNumberLiteral
+	: parameter
+	| INTEGER_LITERAL
+	;
```

Comment:
That obviously requires deeper integration but I think parameters might be nice here. I am imagining to add the limit/offset parameter values to the query cache key if the dbms doesn't support parameters. wdys?